contracts/infrastructure/salary/PayrollClerk.sol

```diff
@@ -56,32 +56,35 @@ contract PayrollClerk is PayrollClerkStorage {
     return workers.length;
   }
 
+  function getPrice(address token) public view returns (uint256) {
+    return IPriceCalculator(calculator()).getPriceWithDefaultOutput(token);
+  }
+
   function multiplePay(address[] calldata _workers, uint256[] calldata _workedHours)
   external onlyControllerOrGovernance {
     require(_workers.length == _workedHours.length, "wrong arrays");
-    for (uint256 i = 0; i < _workers.length; i++) {
-      pay(_workers[i], _workedHours[i]);
+    for (uint256 h = 0; h < tokens.length; h++) {
+      uint256 tPrice = getPrice(tokens[h]);
+
+      for (uint256 i = 0; i < _workers.length; i++) {
+        pay(_workers[i], _workedHours[i], tokens[h], tPrice);
+      }
     }
   }
 
-  function pay(address worker, uint256 _workedHours) public onlyControllerOrGovernance {
+  function pay(address worker, uint256 _workedHours, address token, uint256 tPrice) public onlyControllerOrGovernance {
```

Comment:
change it to `internal` and keep old interface for `pay` function
caller should not setup the price himself

Reply:
I will change it to `internal` and I will get the `tPrice` variable inside the function.

However to keep the old interface of `pay` I would have to set a restriction to only use it when there is one token. And to be able to pay with multiple tokens would then require another function with some additional checking in the `multiplePay` function.

Reply:
I added the params back and updated the tests to use `multiplePay`. Now 2 tests are failing and I haven't figured out why that is.